velox/functions/lib/Slice.cpp

```diff
@@ -48,6 +48,8 @@ namespace {
 /// After adjustment, for the output ArrayVector:
 /// rawOffsets vector [1, 4, 8]
 /// rawSizes vector   [2, 2, 2]
+/// @tparam Kind The type kind of start and length.
```

Comment:
This template parameter is used only once outside of the per-row loop. Hence, there is no perf benefit in using template parameter, but it increases the binary size quite a bit. Consider using constructor parameter instead.

Also, I see `applyArray<int64_t>` which implies that start and length are 64-bit integers. Not sure why this happens to work, but it feels hacky.